src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,44 @@
 package core.mate.academy.service;
 
+import core.mate.academy.machine.impl.BulldozerProducerImpl;
+import core.mate.academy.machine.impl.ExcavatorProducerImpl;
+import core.mate.academy.machine.impl.TruckProducerImpl;
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<Machine> machineList = new ArrayList<>();
+
+        if (type.equals(Bulldozer.class)) {
+            machineList.addAll(new BulldozerProducerImpl().get());
+        } else if (type.equals(Excavator.class)) {
+            machineList.addAll(new ExcavatorProducerImpl().get());
+        } else if (type.equals(Truck.class)) {
+            machineList.addAll(new TruckProducerImpl().get());
+        }
+        return machineList;
```

Comment:
Keep if-else constructions simple. Since each if condition ends with a return, the else keyword is unnecessary and can be omitted for cleaner code.